Drop host name from navigation bar

martinpitt opened the pull request:
Showing the host name makes no sense in the context of navigating the file system tree. We don't have a concept of remote file systems, it just shows the local one. Also, the host name is already shown in the Shell.

Host names are often very long, taking away precious space especially in the mobile view. We already have an icon to represent the file system root. Add a "File system" ARIA label to that button make it accessible.

----

Looks like this now:

![image](https://github.com/user-attachments/assets/c7f51929-2d29-4087-8221-30856b67a031)

Also see the [updated pixel refs](https://github.com/cockpit-project/pixel-test-reference/compare/2e36736bab8c856cf5dfe2f2d22d321c6cbbaf8d..39d4d3a68b803887c70fca2bff53b490557cbba8). I'm fine with debating/changing the precise label of course.

allisonkarlitskaya reviewed (commented):
Happy to see the giant blob of hook code drop, but I wonder if we can save it.  I considered implementing that as a more generic `useDBusProperty()`.  I feel like that could see use elsewhere...

martinpitt commented:
The pixels are already conflicting. I won't resolve this until after a first review, though -- the pixel diff in the description remains valid.

garrett commented:
Should we use "System" instead of "Filesystem"?

Also, while we're there, can we fix the icon as well? It's supposed to use the `OutlinedHddIcon` / `fa-hdd` icon from PF @ 24 px.

This is what it should look like:

![image](https://github.com/user-attachments/assets/72a04662-fc1d-492b-aa63-79b873171481)

martinpitt commented:
Thanks @garrett  - I'm happy to change the icon. I find "system" rather unspecific -- it doesn't mean much by itself, and you never hear/see "go to the system tree" or so. But it's still better than showing the host name, so if you don't like "file system" I'll change it to that.

allisonkarlitskaya commented:
for comparison: Nautilus uses the `NAME` field from `os-release` for this.  On my system it shows "Fedora Linux".

I'm not sure if you'd consider that a substantial improvement, though, considering where you're coming from.

martinpitt commented:
It's an improvement as it doesn't have a variable length. I'm ok with that as well. I just honestly have trouble understanding why there is apparently so much resistance to calling it the one thing that it is: a file system. :man_shrugging:

garrett commented:
Here's what others do:

# GNOME, Nautilus

As mentioned above, the pretty OS name. I think it's probably showing `NAME=` from `/etc/os-release`.

![image](https://github.com/user-attachments/assets/1389fda1-f117-4b19-af61-0bf148224f5f)


# KDE, Dolphin

Dolphin, from KDE, uses the partition label. On Fedora, it's `fedora_fedora`. Not great.

![image](https://github.com/user-attachments/assets/b3891628-839e-448a-a153-2ab964915c59)

# Windows, Explorer

Windows has an icon at the start of the breadcrumb trail. But you can't click on it. It's useless. Their real top-level directory is "This PC" (once upon a time "My Computer"). Windows, like everyone else, abstracts away everything outside of your home directory, unless you click on "This PC" or a disk in the sidebar. ("This PC" also contains disks too.)

![image](https://github.com/user-attachments/assets/1803e98f-4977-413e-9d68-7c6694c7c051)

# macOS, Finder

macOS doesn't show a breadcrumb at all, unless you right-click on the title. However, you can't right-click on macOS by default, unless you know how (lift all fingers from a "magic mouse" and tap your right on on the right side while not touching the mouse othwise, click and hold for a while, or two finger tap on a "trackpad" (touchpad)). It then shows you a dropdown that is the breadcrumbs in reverse order (deepest on the top, origin on the bottom):

![image](https://github.com/user-attachments/assets/5fac78aa-0f61-4550-8ad0-5131500d2a8b)

If you put macOS into a multi-pane drill-down mode, the toplevel is the hard disk's label, which is "macOS" by default:

![image](https://github.com/user-attachments/assets/281337a0-f1fc-4e7f-8748-2f126ed79395)

But, the machine's name is actually toplevel, which is "Name's Type of Mac" ("Garrett's iMac Pro" in this VM). Machine name and hostname are not the same, but they're not really the same on Linux either; there's a similar split between the pretty computer name and the network "hostname". They're usually similar, but sometimes wildly different.

---

So the various ways to do this are:

1. Generic: File system, Filesystem, System, This PC (what Microsoft does)
3. System name: The pretty name version of the hostname, which is before all the dots and such. (What macOS does.)
4. Hostname: The network hostname. (I think this is what we're currently doing.)
5. Disk label: Whatever the boot partition is called. This can be fine, this can be awful. (What KDE does. Also what macOS does in places.)
6. Operating system name. (What GNOME does.)

We're dropping out both 4 and 3 (hostname and system name). Disk label is bad on Fedora and likely bad on RHEL too, so we're dropping 5. So, we're left with:

- Generic: File system, System
- Operating system name: Fedora Linux, Red Hat Enterprise Linux, Ubuntu, openSUSE Tumbleweed

OK, so what does that look like? I've mocked up all the viable options in this comparison graphic:

![Breadcrumb comparisons](https://github.com/user-attachments/assets/bc691629-e550-4305-badb-e1b180af9d4e)

mac2net commented:
MacOS Finder does show a breadcrumb at the bottom if you turn on `Show Path Bar` from the View Menu.


<img width="230" alt="image" src="https://github.com/user-attachments/assets/53c28f50-2595-4d40-ad65-882b66178109">

<img width="428" alt="image" src="https://github.com/user-attachments/assets/e7d3702c-6846-4745-a6ee-863ab92a0482">

garrett commented:
> We should perhaps avoid the sole icon for a11y reasons (but it's just a gut feeling).

As long as it has a tooltip or ARIA, it's fine accessibility-wise.

garrett commented:
- Having it as a disk label is bad as well, because `sys`, `proc`,  `run` and `tmp` are there and are notably not on the disk.
- Having it as the distro name gets  a bit long with some, notably RHEL.
- Root, as suggested elsewhere, is also bad as it's jargon and there's a folder called "root" too.

The problem that we're hitting is that basically everything is wrong, unless it's the system's name, but then that has the problem we're hitting with being too long.

System and File system are probably the least wrong, all considering?

martinpitt commented:
@garrett @jelly After our discussion here and on Matrix, I updated that to drop the string from the UI and instead gave the icon button an aria-label. a11y wise this seems to make more sense to me than reading "icon button" and the host name. Screenshot and pixel tests updated. I also added as second commit for changing the icon, [see comment above](https://github.com/cockpit-project/cockpit-files/pull/643#issuecomment-2228497671)

martinpitt commented:
Ah, forgot to make the icon bigger. A naive wrapping in ` <Icon size="lg">` doesn't center:

![image](https://github.com/user-attachments/assets/c90895c3-40ca-459b-9e76-f2e89bb98162)

So that will end up in another CSS fight. I'm happy to do this in a follow-up.

martinpitt commented:
Ah, there are a [couple of extra](https://cockpit-logs.us-east-1.linodeobjects.com/pull-643-1e9f6744-20240717-102203-fedora-40/log.html) pixel changes as the file icon area gets bigger now that the breadcrumbs only take one line instead of two. If this is otherwise fine for you, I'll update the refs accordingly.

garrett commented:
> So that will end up in another CSS fight. I'm happy to do this in a follow-up.

The pathbar (breadcrumbs) should be changing soon, in https://github.com/cockpit-project/cockpit-files/issues/640 and https://github.com/cockpit-project/cockpit-files/issues/641.

We're probably fighting against the currently improper implementation at the moment. It should hopefully be simple after those changes land (moving the button outside of the path, and using the breadcrumbs component from PF with a little override to change `>` to `/`).

martinpitt commented:
THis got folded into #670
